### PR TITLE
Fix a build error some users are experiencing wherein __NR_getcpu is not defined

### PR DESCRIPTION
**Pull request by AshkanAliabadi**
https://github.com/pytorch/cpuinfo/issues/36

### CODE REVIEW

File: src/api.c [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 	#include <unistd.h>
 	#include <sys/syscall.h>
+	#include <asm-generic/unistd.h>
```
Comment (pearu) [inline src/api.c:13]: To minimize possible conflicts for systems where inclusion of `unistd.h` already defines `__NR_getcpu`, I suggest:
```
	#if !defined(__NR_getcpu)
		#include <asm-generic/unistd.h>
	#endif
```
With this change, I confirm that pytorch master builds successfully here (Ubuntu 18.04, using conda environment).

**Comment by rgommers** [comment]
In addition to what @pearu said, I would change line 314 from `#ifdef __linux__` to:
```
if defined(__linux__) && defined(__NR_getcpu)
```
That will fix things on older kernels that don't have `__NR_getcpu` at all as well.

**Comment by Maratyszcza** [comment]
I second @pearu's suggestion to include `asm-generic/unistd.h` only if the `unistd.h` doesn't have the right symbol. However, the suggestion in @rgommers's comment is very dangerous: it introduce the risk of silently regressing performance (because `cpuinfo_get_current_*` functions return default result) without any indication of failure. `getcpu` syscall was introduced in Linux 2.6.19 released on Nov 29, 2006 so we can just assume that it is always supported.

File: src/api.c (hunk 1 above)
Comment (AshkanAliabadi) [reply src/api.c:13]: Thanks.  Regarding your question on what PyTorch tests to run - it is difficult to say.  cpuinfo is used both directly inside PyTorch and by its dependencies.  Many codepaths in PyTorch are conditionally enabled or disabled with CMake variables and preprocessor directives in the code, and there are also different platforms to consider so running the CI in its entirety is a must in my opinion.

**Comment by Maratyszcza** [comment]
LGTM

**Comment by rgommers** [comment]
> However, the suggestion in @rgommers's comment is very dangerous: it introduce the risk of silently regressing performance

I don't think that's true, since it can only fail if the symbol is missing completely, i.e. for old kernels. Also, pretty much every use of `__NR_getcpu` in public code (including in Tensorflow [here](https://github.com/tensorflow/tensorflow/blob/fa3b55e940cef475c8691487f4c00ac900a67d04/tensorflow/core/platform/default/port.cc#L100)) is protected like that. Within 2 weeks of merging the unprotected version, multiple people ran into build issues. 

That said, there was no response yet from @bryan-lunt on gh-36, and this does fix things for @pearu and me, so merging this PR as-is is an improvement. It's still likely to come back as runtime errors once people on older clusters install the next release though.

**Comment by AshkanAliabadi** [comment]
@driess Can you please stamp this?

**Comment by rgommers** [comment]
A ping on this, can someone merge this PR please?

**Comment by rgommers** [comment]
Thanks! PR to update submodule in PyTorch at https://github.com/pytorch/pytorch/pull/37083

**Pull request merged**